### PR TITLE
#3348 implement file list, upload, download and delete

### PR DESCRIPTION
**Pull request by eechih**
### Checklist
- [x] [Review Contribution Guidelines](https://ohara.readthedocs.io/en/latest/contributing.html)
- [x] Add Reviewers (see [Authors](https://github.com/oharastream/ohara#ohara-team))
- [x] Add Assignees (of course, you are the assignee by default!)
- [x] DON'T set a Milestone
- [x] DON'T add any labels (PR labels will be added automatically by Jenkins)
- [x] Link to an issue(You can do so by using the required by keyword: required by #3348 )
- [ ] Add unit tests (or please explain why this PR is not worth having new tests)
- [ ] Pass QA (Sometimes you may encounter an existing flaky test which would obstruct you from getting QA passed. If the failed tests are unrelated, you can add a comment in the PR thread)

### CODE REVIEW

File: ohara-manager/client/src/components/common/Search/QuickSearch.js [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+  const search = useCallback(createSearch(keys), [keys]);
+
+  useEffect(() => {
+    if (data === prevData && searchText === prevSearchText) return;
+    if (!searchText) setResults(data);
+    else setResults(search(data, searchText));
+  }, [data, prevData, searchText, prevSearchText, search, setResults]);
+
+  const handleChange = debounce(
```
Comment (konekoya) [inline ohara-manager/client/src/components/common/Search/QuickSearch.js]: Well done @eechih  I couldn't get debounce work with Hooks without a custom hook last week. Will give it another shot :)

File: ohara-manager/client/src/components/common/Search/QuickSearch.js [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+import InputBase from '@material-ui/core/InputBase';
+import IconButton from '@material-ui/core/IconButton';
+import SearchIcon from '@material-ui/icons/Search';
+import { usePrevious } from 'utils/hooks';
+import { createSearch } from './search';
+
+export const Wrapper = styled.div(
+  ({ theme }) => css`
+    .MuiPaper-root {
+      padding: '${theme.spacing(0.5)}px ${theme.spacing(1)}px';
```
Comment (konekoya) [inline ohara-manager/client/src/components/common/Search/QuickSearch.js]: I think you can also do `'${theme.spacing(0.5, 1)}`

File: ohara-manager/client/src/components/Workspace/File/FileUpload.js [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+const FileUpload = ({ children, accept = '.jar' }) => {
+  const { currentWorkspace } = useWorkspace();
+  const { uploadFile } = useFileActions();
+  const inputEl = useRef(null);
+
+  const Trigger = () => {
+    const handleClick = () => inputEl.current.click();
+
+    if (children) {
+      return cloneElement(children, {
```
Comment (konekoya) [inline ohara-manager/client/src/components/Workspace/File/FileUpload.js]: This usage is very interesting. should we leave some comment here :)

Comment (eechih) [reply ohara-manager/client/src/components/Workspace/File/FileUpload.js]: No problem. I will comment it  :)

**Pull request merged**